### ISSUE TITLE
Idea: Relaton toolchain

### ISSUE THREAD

**Issue by abunashir** [relaton/api.relaton.org#1]
## Idea

We are planning to do some major changes in the relaton toolchain, and the main idea with this is to have everything centralized as a service and have the relaton library communicate with this. So, the major components are

* Relaton Service / API
* Relaton Library communicating with the API
* Relaton CLI

### Relaton API / Service:

The API/Service is an application that will work as a hub of everything, and it will take care of all the heavy lifting and provide a clear API for the user. some of the major tasks it will handles are

- [x] Fetch standard documents from supported sources
- [x] Parse this fetched documents to the relaton model
- [x] Maintain a global documents cache (complying with copyright)
- [ ] API endpoints to serve all document request
- [ ] Web interface that allows a user to browse/maintain documents

### Relaton Library

The relaton library is mainly responsible for communicating with the Relaton API, Here are some of the major task it should handle.

* Communicating with service API and serialize the response
* Maintain a local cache for the user's requested documents

### Relaton CLI

The relaton CLI would be something that wraps around the relaton library, and also provide some common utility tasks. This could also include some functionalities that we are doing suing some makefile at the moment.

cc: @ronaldtse

**Comment by ronaldtse** [comment]
Agreed. We will still keep the data-source fetching gems open source so that the data source organizations (or us) can keep them updated.

**Comment by abunashir** [comment]
@ronaldtse: In the recent changes,  I've added the fetch interface, and as we discussed it parses the content, and maintain a temporary cache and serve all API request using our service.

What are the next step you would like to have in the API?

**Comment by ronaldtse** [comment]
From @abunashir :

* Adopt/Implement this new service API to the relaton library
* Extract existing standard parsing/fetching from main source to a separate library
* Update the caching mechanism, and use the new json response to build those object
* Add endpoint to list all available standards
* Build the website to show all of this as service
* Build graphql endpoint for all of this existing behavior

**Comment by ronaldtse** [comment]
The Relaton site is meant to be an aggregation site of all standards and their relationships. We should provide a browser view and also API access. 

In terms of priority, we want to:

1. The relaton api should be able to handle all current flavors of standards (the `relaton-*` gems)
2. The relaton user library should be able to fetch from the Relaton API
3. For each relaton bibliography entry, we should provide a stable URI for referencing, and a browser view of it, using content-type or mime. (API + Browser viewable)
4. Then we enhance the metadata of the standards to build standards relationships. 

> @abunashir: Another thing, if we update the relaton user library, how might that affect existing tools? apart from relaton-cli, and who else is our user for relaton library?

Our major user is Metanorma, which is used by SDOs. You can make the experience transparent by working with @opoudjis , @andrew2net and the other guys.
